proto/redpanda/core/admin/v2/shadow_link.proto

```diff
@@ -249,6 +249,8 @@ message SecuritySettingsSyncOptions {
 }
 // TLS settings
 message TLSSettings {
+    // Whether or not TLS is enabled
+    bool enabled = 3;
```

Comment:
quesiton:  shloud tls_settings be optional in this case? (that expresses intent that even if enabled = true, the settings are optional).

Reply:
We could have added system TLS as an additional oneof instead of a boolean flag that causes the oneof to be ignored at times.

```
message System {}
oneof tls_setting {
  System system = 1;
  ...
}
```